core/src/main/scala/org/apache/spark/status/api/v1/StagesResource.scala

```diff
@@ -189,32 +189,41 @@ private[v1] class StagesResource extends BaseAppResource {
     val taskMetricsContainsValue = (task: TaskData) => task.taskMetrics match {
       case None => false
       case Some(metrics) =>
-        (containsValue(task.taskMetrics.get.executorDeserializeTime)
-        || containsValue(task.taskMetrics.get.executorRunTime)
-        || containsValue(task.taskMetrics.get.jvmGcTime)
-        || containsValue(task.taskMetrics.get.resultSerializationTime)
-        || containsValue(task.taskMetrics.get.memoryBytesSpilled)
-        || containsValue(task.taskMetrics.get.diskBytesSpilled)
-        || containsValue(task.taskMetrics.get.peakExecutionMemory)
-        || containsValue(task.taskMetrics.get.inputMetrics.bytesRead)
+        (containsValue(UIUtils.formatDuration(task.taskMetrics.get.executorDeserializeTime))
+        || containsValue(UIUtils.formatDuration(task.taskMetrics.get.executorRunTime))
+        || containsValue(UIUtils.formatDuration(task.taskMetrics.get.jvmGcTime))
+        || containsValue(UIUtils.formatDuration(task.taskMetrics.get.resultSerializationTime))
+        || containsValue(UIUtils.formatBytes(task.taskMetrics.get.memoryBytesSpilled))
+        || containsValue(UIUtils.formatBytes(task.taskMetrics.get.diskBytesSpilled))
+        || containsValue(UIUtils.formatBytes(task.taskMetrics.get.peakExecutionMemory))
+        || containsValue(UIUtils.formatBytes(task.taskMetrics.get.inputMetrics.bytesRead))
         || containsValue(task.taskMetrics.get.inputMetrics.recordsRead)
-        || containsValue(task.taskMetrics.get.outputMetrics.bytesWritten)
+        || containsValue(UIUtils.formatBytes(task.taskMetrics.get.outputMetrics.bytesWritten))
         || containsValue(task.taskMetrics.get.outputMetrics.recordsWritten)
-        || containsValue(task.taskMetrics.get.shuffleReadMetrics.fetchWaitTime)
+        || containsValue(UIUtils.formatDuration(
+          task.taskMetrics.get.shuffleReadMetrics.fetchWaitTime))
+        || containsValue(UIUtils.formatBytes(
+          task.taskMetrics.get.shuffleReadMetrics.remoteBytesRead))
+        || containsValue(UIUtils.formatBytes(
+          task.taskMetrics.get.shuffleReadMetrics.localBytesRead +
+          task.taskMetrics.get.shuffleReadMetrics.remoteBytesRead))
         || containsValue(task.taskMetrics.get.shuffleReadMetrics.recordsRead)
-        || containsValue(task.taskMetrics.get.shuffleWriteMetrics.bytesWritten)
+        || containsValue(UIUtils.formatBytes(
+          task.taskMetrics.get.shuffleWriteMetrics.bytesWritten))
         || containsValue(task.taskMetrics.get.shuffleWriteMetrics.recordsWritten)
-        || containsValue(task.taskMetrics.get.shuffleWriteMetrics.writeTime))
+        || containsValue(UIUtils.formatDuration(
+          task.taskMetrics.get.shuffleWriteMetrics.writeTime / 1000000)))
```

Comment:
I think that should be `1000`. Unless `shuffleWriteMetrics` are in kiloseconds :) 

`formatDuration` is expecting milliseconds.

Reply:
So basically, `shuffleWriteMetrics` is recorded in nanoseconds. That is why, I am dividing the number by `1000000`. Have attached screenshot for reference.

<img width="369" alt="Screen Shot 2019-04-22 at 2 12 36 PM" src="https://user-images.githubusercontent.com/22228190/56519923-c6be1300-6508-11e9-8171-57a747cb68ad.png">

Reply:
You can find that a similar operation has also been performed in stagepage.js on `writeTime`.